### PR TITLE
MEPTS-58: DataBaseService

### PR DESCRIPTION
**Pull request by edrisse**

### CODE REVIEW

**Comment by nthfloor** [comment]
@edrisse the build is failing and codacy is complaining please take a look.

File: src/main/java/com/openmrs/migrator/core/services/CommandService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.openmrs.migrator.core.services;
+
+
```
Comment (nthfloor) [inline src/main/java/com/openmrs/migrator/core/services/CommandService.java]: Please some comment blocks at the top of each service to indicate the purpose of each service

File: src/main/java/exception/CommandExecutionException.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package exception;
```
Comment (nthfloor) [inline src/main/java/exception/CommandExecutionException.java]: please move this folder under the core folder

so package should be: `migrator.core.exceptions`

File: src/main/java/com/openmrs/migrator/core/services/impl/DBServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		    "-h" + getDatabasePassword(), "-e", String.format("use %s; source %s;", databaseName, fileName));
+	}
+	
+	@Override
+	public void createDatabase(String databaseName) {
+		commandService.runCommand("mysql", "-u" + getDatabaseUser(), "-p" + getDatabasePassword(), "-h" + getDatabaseHost(),
+		    "-e", String.format("drop database if exists %s; create database %s;", databaseName, databaseName));
+	}
+	
+	private String getDatabaseHost() {
```
Comment (nthfloor) [inline src/main/java/com/openmrs/migrator/core/services/impl/DBServiceImpl.java]: Please move this to the ConfigurationService - no point in keeping it here. We can leave them like this as hard coded values there.

File: src/main/java/com/openmrs/migrator/core/services/impl/DBServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import com.openmrs.migrator.core.services.DBService;
+
+@Service
+public class DBServiceImpl implements DBService {
+	
+	@Autowired
+	private CommandService commandService;
+	
+	@Override
+	public void importDatabaseFile(String databaseName, String fileName) {
```
Comment (nthfloor) [inline src/main/java/com/openmrs/migrator/core/services/impl/DBServiceImpl.java]: Can we add some unit tests for these?

File: src/main/java/com/openmrs/migrator/core/services/impl/ComandServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import exception.CommandExecutionException;
+
+@Service
+public class ComandServiceImpl implements CommandService {
+	
+	private static Logger LOG = LoggerFactory.getLogger(ComandServiceImpl.class);
+	
+	@Override
+	public void runCommand(String... args) {
```
Comment (nthfloor) [inline src/main/java/com/openmrs/migrator/core/services/impl/ComandServiceImpl.java]: can we add some unit tests for this method?

**Review by nthfloor — CHANGES_REQUESTED** [review]
@edrisse looking good, just some small comments

File: src/test/java/com/openmrs/migrator/core/services/CommandServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CommandServiceTest {
+
+  @Autowired private CommandService commandService;
+
+  @Test
+  public void shouldExecuteLsCommand() {
+    try {
+      commandService.runCommand("ls", "-ltr");
```
Comment (nthfloor) [inline src/test/java/com/openmrs/migrator/core/services/CommandServiceTest.java:20]: Is it possible to hide the result from this command? When I run the tests it displays the list of directories in the current folder...

Comment (edrisse) [reply src/test/java/com/openmrs/migrator/core/services/CommandServiceTest.java:20]: Done

**Pull request merged**